apps/dashboard/app/(app)/[workspaceSlug]/settings/billing/components/plan-selection-modal.tsx

```diff
@@ -0,0 +1,280 @@
+"use client";
+
+import { formatNumber } from "@/lib/fmt";
+import { trpc } from "@/lib/trpc/client";
+import { cn } from "@/lib/utils";
+import { Button, DialogContainer, toast } from "@unkey/ui";
+import { useRouter } from "next/navigation";
+import { useCallback, useEffect, useState } from "react";
+
+type PlanSelectionModalProps = {
+  isOpen: boolean;
+  onOpenChange?: (open: boolean) => void;
+  products: Array<{
+    id: string;
+    name: string;
+    priceId: string;
+    dollar: number;
+    quotas: {
+      requestsPerMonth: number;
+    };
+  }>;
+  workspaceSlug: string;
+  currentProductId?: string;
+  isChangingPlan?: boolean;
+};
+
+export const PlanSelectionModal = ({
+  isOpen,
+  onOpenChange,
+  products,
+  workspaceSlug,
+  currentProductId,
+  isChangingPlan = false,
+}: PlanSelectionModalProps) => {
+  const [selectedProductId, setSelectedProductId] = useState<string | null>(null);
+  const [isLoading, setIsLoading] = useState(false);
+  const [hasMounted, setHasMounted] = useState(false);
+  const router = useRouter();
+  const trpcUtils = trpc.useUtils();
+
+  // Initialize selectedProductId after mount to prevent hydration mismatch
+  useEffect(() => {
```

Comment:
are we still getting that hydration mismatch after calling this with dynamic import and SSR: false?